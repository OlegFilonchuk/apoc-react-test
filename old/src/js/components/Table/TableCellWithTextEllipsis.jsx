import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import TableCell from './TableCell';
import ExpandButton from './ExpandButton';
import MultilineEllipsisPopover from '../MultilineEllipsisPopover/MultilineEllipsisPopover';
import POPOVER_PLACEMENTS from '../Popover/PopoverPlacements';

import './TableCellWithTextEllipsis.less';

/* eslint-disable */

const baseClassName = 'td-cell-with-text-ellipsis';

export default class TableCellWithTextEllipsis extends React.PureComponent {
  render() {
    const { children, className, hideAfterDelayOf, maxLines, popoverMaxWidth, ...restOfProps } = this.props;
    const arrayOfChildren = React.Children.toArray(children);
    const content = arrayOfChildren.filter(child => child.type !== ExpandButton);
    const expandButton = arrayOfChildren.find(child => child.type === ExpandButton);

    return (
      <TableCell {...restOfProps} className={classNames(className, baseClassName)}>
        {expandButton}
        <MultilineEllipsisPopover
          maxLines={maxLines}
          popoverMaxWidth={popoverMaxWidth}
          hideAfterDelayOf={hideAfterDelayOf}
        >
          {content}
        </MultilineEllipsisPopover>
      </TableCell>
    );
  }
}

TableCellWithTextEllipsis.propTypes = {
  /**
   * TableCell's propTypes
   */
  ...TableCell.propTypes,
  /**
   * Cell's max height
   */
  maxLines: PropTypes.number,

  /**
   * This function will be called click outside of popover element
   */
  placement: PropTypes.oneOf(Object.keys(POPOVER_PLACEMENTS).map(placement => POPOVER_PLACEMENTS[placement])),
  /**
   * Expects time in milliseconds
   * After this time popover will be hidden if there was no onMouseOver event
   */
  hideAfterDelayOf: PropTypes.number,
  /**
   * Cell's content
   */
  children: PropTypes.node,

  /**
   * List of custom classes added to component
   */
  className: PropTypes.string,

  /**
   * Cell's width
   */
  width: PropTypes.string,

  /**
   *  popoverMaxWidth
   */
  popoverMaxWidth: PropTypes.number
};

TableCellWithTextEllipsis.defaultProps = {
  children: null,
  className: '',
  width: '',
  hideAfterDelayOf: 0,
  maxLines: 3,
  placement: POPOVER_PLACEMENTS.TOP_LEFT,
  popoverMaxWidth: undefined
};
