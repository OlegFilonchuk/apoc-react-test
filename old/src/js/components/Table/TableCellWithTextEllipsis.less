@import './tableVariables.less';

.td-cell-with-text-ellipsis {
  background-color: inherit;
}
