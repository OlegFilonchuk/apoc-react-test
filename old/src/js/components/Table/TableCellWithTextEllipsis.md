### Example of table with td with overflow content
It uses [TableHeadRow](#tableheadrow) and [TableCellWithTextEllipsis](#TableCellWithTextEllipsis)

    const initialState = { 
      text: 'Lorem ipsum. Almost truncated. Almost truncated. Almost truncatee truncatee truncteee'
    };
    
    const onChange = e => setState({text:e.target.value});
   
    
    (
      <div>     
        <br/>      
        <Table isRowsStriped>
          <TableHeadRow>
            <th>Max pages</th>
            <th>Text</th>
          </TableHeadRow>
          <tbody>            
            <tr>              
              <td width="500px">Next cell into right is ellipsed after 2 lines</td>
              <TableCellWithTextEllipsis key={state.step} width="250px" maxLines={2}>
                {state.text}
              </TableCellWithTextEllipsis>
            </tr>                                                
            <tr>              
              <td width="500px">Next cell into right is ellipsed after 1 line</td>
              <TableCellWithTextEllipsis key={state.step} width="250px" maxLines={1}>
                {state.text}
              </TableCellWithTextEllipsis>
            </tr>
            <tr>              
              <td width="500px">Next cell into right is ellipsed after 3 lines</td>
              <TableCellWithTextEllipsis key={state.step} width="250px" maxLines={3}>
                {state.text}
              </TableCellWithTextEllipsis>
            </tr>           
          </tbody>
        </Table>
        <div>
          <p><b>Edit text in cells</b></p>
          <textarea cols="50" rows="3" onChange={onChange} value={state.text}/>
        </div>
      </div>
    )
    
### Comparision of usage `td` + [MultilineEllipsisPopover](#multilineellipsispopover)

    (
          <div>     
            <br/>      
            <Table isRowsStriped>
              <TableHeadRow>
                <th>Max pages</th>
                <th>Text</th>
              </TableHeadRow>
              <tbody>            
                <tr>              
                  <td width="500px">1 line with `TableCellWithTextEllipsis`</td>
                  <TableCellWithTextEllipsis key={state.step} width="250px" maxLines={1}>
                    Shipmates sing from courages like evil fishs. Gar, fine sea. go to haiti.
                  </TableCellWithTextEllipsis>
                </tr>                                                
                <tr>              
                  <td width="500px">1 line with `td` + `MultilineEllipsisPopover`</td>
                  <td key={state.step} width="250px">
                    <MultilineEllipsisPopover maxLines={1}>
                      Shipmates sing from courages like evil fishs. Gar, fine sea. go to haiti.
                    </MultilineEllipsisPopover>
                  </td>
                </tr>                           
              </tbody>
            </Table>          
          </div>
        )
