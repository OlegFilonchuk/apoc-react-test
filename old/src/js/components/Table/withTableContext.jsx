import React from 'react';
import propTypes from 'prop-types';

export const CONTEXT_ID = 'APOC_REACT_TABLE_CONTEXT';

export const tableContextPropTypes = propTypes.shape({
  /**
   * @type {function(uuid: string, index:number, contentTypes: CellContentTypes)}
   */
  registerColumn: propTypes.func,
  /**
   * @type {function(uuid: string)}
   */
  unregisterColumn: propTypes.func,
  /**
   * @type {function(index: ?number):CellContentTypes}
   */
  getColumnContentTypes: propTypes.func
});

/**
 * @type {CellContentTypes}
 */
export const defaultContentTypes = {
  dateContent: false,
  iconOrGraphicalContent: false,
  textContent: false,
  numericContent: false
};

export const defaultCellTableProps = {
  registerColumn: () => null,
  unregisterColumn: () => null,
  getColumnContentTypes: () => defaultContentTypes
};

export function withTableContext(Component) {
  function WithTableContext(props, context) {
    return <Component {...props} tableProps={context[CONTEXT_ID]} />;
  }

  WithTableContext.propTypes = Component.propTypes;

  WithTableContext.contextTypes = {
    [CONTEXT_ID]: tableContextPropTypes
  };

  WithTableContext.displayName = `withTableContext${Component.displayName || Component.name}`;

  WithTableContext.WrappedComponent = Component;

  return WithTableContext;
}

/* eslint-disable no-unused-vars */
/**
 * @typedef {{dateContent: boolean,iconOrGraphicalContent: boolean,textContent: boolean, numericContent: boolean}} CellContentTypes
 */
const CellContentTypes = {};
/* eslint-enable no-unused-vars */
