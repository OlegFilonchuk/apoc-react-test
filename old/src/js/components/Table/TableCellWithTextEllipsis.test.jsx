import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';

import TableCellWithTextEllipsis from './TableCellWithTextEllipsis';
import { defaultCellTableProps, CONTEXT_ID } from './withTableContext';

const mountCell = ({ children, ...restOfProps }) =>
  mount(<TableCellWithTextEllipsis {...restOfProps}>{children || 'Hello world'}</TableCellWithTextEllipsis>, {
    attachTo: document.createElement('tr'),
    context: {
      [CONTEXT_ID]: defaultCellTableProps
    }
  });

/* eslint-disable global-require, no-return-assign */
describe('<TableCellWithTextEllipsis />', () => {
  describe('is not ellipsed', () => {
    let element;

    beforeEach(() => (element = mountCell({})));

    it('matches snapshot', () => {
      expect(mountToJson(element)).toMatchSnapshot();
    });
  });

  describe('with className prop set', () => {
    let element;

    beforeEach(() => (element = mountCell({ className: 'some-extra-class' })));
    it('correctly set `className`', () => {
      const td = element.find('td');

      expect(td).toHaveClassName('some-extra-class');
    });
  });

  describe('with with prop set', () => {
    let element;

    beforeEach(() => (element = mountCell({ width: '100px' })));
    it('correctly set `className`', () => {
      const td = element.find('td');

      expect(td).toHaveStyle('width', '100px');
      expect(td).toHaveStyle('maxWidth', '100px');
    });
  });
});
/* eslint-enable global-require, no-return-assign */
