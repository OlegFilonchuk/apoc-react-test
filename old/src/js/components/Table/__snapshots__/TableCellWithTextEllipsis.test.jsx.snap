// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableCellWithTextEllipsis /> is not ellipsed matches snapshot 1`] = `
<TableCellWithTextEllipsis
  className=""
  hideAfterDelayOf={0}
  maxLines={3}
  placement="topLeft"
  popoverMaxWidth={undefined}
  width=""
>
  <withTableContextTableCell
    className="td-cell-with-text-ellipsis"
    placement="topLeft"
    width=""
  >
    <TableCell
      className="td-cell-with-text-ellipsis"
      placement="topLeft"
      tableProps={
        Object {
          "getColumnContentTypes": [Function],
          "registerColumn": [Function],
          "unregisterColumn": [Function],
        }
      }
      width=""
    >
      <td
        className="td-cell-with-text-ellipsis sc-center-align"
        placement="topLeft"
        style={Object {}}
      >
        <MultilineEllipsisPopover
          hideAfterDelayOf={0}
          maxLines={3}
          placement="topLeft"
          popoverMaxWidth={undefined}
        >
          <MultilineEllipsis
            innerRef={[Function]}
            maxLines={3}
          >
            <_class
              client={true}
              innerRef={[Function]}
              onResize={[Function]}
              scroll={true}
            >
              <Measure
                client={true}
                contentRect={
                  Object {
                    "bounds": Object {},
                    "client": Object {},
                    "entry": Object {},
                    "margin": Object {},
                    "offset": Object {},
                    "scroll": Object {},
                  }
                }
                measure={[Function]}
                measureRef={[Function]}
                scroll={true}
              >
                <div
                  key="truncated"
                >
                  Hello world
                </div>
              </Measure>
            </_class>
          </MultilineEllipsis>
        </MultilineEllipsisPopover>
      </td>
    </TableCell>
  </withTableContextTableCell>
</TableCellWithTextEllipsis>
`;
